### PR TITLE
NAVAND-979: support yards and ios/android thresholds parity

### PR DESCRIPTION
**Pull request by dzinad**
This PR:
1. Uses yards for imperial UK distances;
2. Supports android/iOS parity for distance thresholds (https://github.com/mapbox/mapbox-navigation-ios/blob/96d0fc0744d64d2a092f571e1583eeb86831a774/Sources/MapboxCoreNavigation/DistanceFormatter.swift#L82).

Rounding increment parity is not supported here, it will be addressed separately.

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt [hunk 1]
```diff
@@ -39,38 +36,204 @@ object MapboxDistanceUtil {
...
+            UnitType.IMPERIAL -> {
+                val distanceInMiles = TurfConversion.convertLength(
+                    distanceInMeters,
+                    TurfConstants.UNIT_METERS,
+                    TurfConstants.UNIT_MILES
+                )
+                if (locale.language == enLanguage && locale.country == "GB") {
+                    getUKDistance(distanceInMiles, roundingIncrement, locale)
+                } else {
+                    getUSDistance(distanceInMiles, roundingIncrement, locale)
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:72]: I'm not really an expert, but should we take into account other locations (like Australia and maybe some else)? Maybe set a more generic name for this? like `getYardDistance`. 
Also, do we really need to check the language for the GB locale?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt [hunk 2]
```diff
@@ -105,132 +268,44 @@ object MapboxDistanceUtil {
      * @param unitType indicates whether the value should be returned as metric or imperial
      * @return an object containing values for displaying the formatted distance
      */
+    @JvmOverloads
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:271]: wdyt if change the signature to wrapper (builder included) to support future possible changes?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt [hunk 3]
```diff
@@ -39,38 +36,204 @@ object MapboxDistanceUtil {
...
+                UnitType.METRIC
+            )
+            else -> {
+                val distanceInKm = TurfConversion.convertLength(
+                    distanceInMeters,
+                    TurfConstants.UNIT_METERS,
+                    TurfConstants.UNIT_KILOMETERS
+                )
+                when {
+                    distanceInMeters < 3000.0 -> largeValue(
```
[the remark this answers is not in the document]
Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:103]: Android/iOS parity.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt (hunk 1 above)
Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:72]: That's a good question. In the ticket only UK was being discussed.
@abhishek1508 @1ec5 are there any other countries that we should use yards in? As far as I understood, on iOS it's just GB, right?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt (hunk 2 above)
Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:271]: Do you mean introduce a new method like
`formatDistance(Wrapper)`
where `Wrapper` is a model class with builder?
I'm not sure. I mean if we add a new parameter we will probably need it in our calculations. Which means that we will have to provide the default value anyway to be backward compatible. How is Builder going to help here?

Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:271]: java developers cannot fully rely on default parameters. For instance in the method 
```kotlin
@JvmOverloads
fun method(
  a: A,
  b: B = defaultB,
  c: C = defaultC
)
```

java developers cannot overload the default value for param `b`, and keep the same for `c`, so invoking of method on the java side would look like (we're forcing to set param `c`, even if default value exists)

```java
#method(a, customValuesB, defaultC)
```

Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:271]: I think you meant the other way around.
But OK, agreed that default parameters may be tricky to support in the future so I'll add a new method just the way it's done for the `formatDistance` that returns `FormattedDistanceData`.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt (hunk 1 above)
Comment (dzinad) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:72]: Checked that on iOS the logic is exactly the same: https://github.com/mapbox/mapbox-navigation-ios/blob/96d0fc0744d64d2a092f571e1583eeb86831a774/Sources/MapboxCoreNavigation/DistanceFormatter.swift#L140

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:103]: as soon as the PR will be merged, it's lost the context of these values. Could you move values to `const` and add comments 🙏

[the remark this answers is not in the document]
Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt:103]: I think so 👍 IMO the same important make leave the comments that the values are in sync with iOS

**Review by VysotskiVadim — APPROVED** [review]

**Pull request merged**